src/resource/nonRdfData.test.ts

```diff
@@ -38,10 +38,57 @@ import {
   saveFileInContainer,
   overwriteFile,
   getFileWithAcl,
+  flattenHeaders,
 } from "./nonRdfData";
 import { Headers, Response } from "cross-fetch";
 import { WithResourceInfo } from "../interfaces";
 
+describe("flattenHeaders", () => {
+  it("returns an empty object for undefined headers", () => {
+    expect(flattenHeaders(undefined)).toEqual({});
+  });
+
+  it("returns well-formed headers as-is", () => {
+    const headers: Record<string, string> = {
+      test: "value",
+    };
+    expect(flattenHeaders(headers)).toEqual(headers);
+  });
+
+  it("transforms an incoming Headers object into a flat headers structure", () => {
+    const myHeaders = new Headers();
+    myHeaders.append("accept", "application/json");
+    myHeaders.append("content-type", "text/turtle");
+    // The following needs to be ignored because `node-fetch::Headers` and
+    // `lib.dom.d.ts::Headers` don't align. It doesn't break the way we
+    // use them currently, but it's something that must be cleaned up
+    // at some point.
+    // eslint-disable-next-line
+    // @ts-ignore
+    const flatHeaders = flattenHeaders(myHeaders);
+    expect(Object.entries(flatHeaders)).toEqual([
+      ["accept", "application/json"],
+      ["content-type", "text/turtle"],
+    ]);
```

Comment:
Is there a reason you're converting this back to `Array<[string, string]>`, rather than

```suggestion
    expect(flatHeaders).toEqual({
      "accept", "application/json",
      "content-type", "text/turtle",
    });
```

?

Reply:
Maybe there was, but I don't remember what it was, and the alternative looks better, fixed in 1c86607ffeae8c20582177afbf4aba6b0021e586

Reply:
:+1: two more places were the same applies I suppose :)